### ISSUE TITLE
how to build the dynamic_dispatch example ?

### ISSUE THREAD

**Issue by kforner** [p12tic/libsimdpp#28]
Hello,
Probably a silly question, but since there is absolutely no install or basic usage documentation...

So I got the git repo, then:
```
cd examples/dynamic_dispatch
make test
```

and I get:
```
In file included from test.cc:4:0:
../../simdpp/dispatch/get_arch_gcc_builtin_cpu_supports.h: In function ‘simdpp::Arch simdpp::get_arch_gcc_builtin_cpu_supports()’:
../../simdpp/dispatch/get_arch_gcc_builtin_cpu_supports.h:24:41: error: Parameter to builtin not valid: avx512f
     if (__builtin_cpu_supports("avx512f")) {

```

```

gcc (Ubuntu 4.8.4-2ubuntu1~14.04) 4.8.4

```

**Comment by p12tic** [comment]
Thanks for finding this. Essentially, that part of the library has not been tested with older compilers. As a work around you can comment out that if block.

A fix is being prepared already.

**Comment by p12tic** [comment]
Fixed in d67594771e88b4cf8bb.

**Comment by kforner** [comment]
Does not seem to work for me:
```
%make

g++ main.cc -std=c++11 -I../../ -c -o main.o
g++ test.cc -c -std=c++11 -I../../ -DSIMDPP_ARCH_X86_SSE2 -msse2 \
        -DSIMDPP_EMIT_DISPATCHER=1 \
        -DSIMDPP_DISPATCH_ARCH1=SIMDPP_ARCH_X86_SSE2 \
        -DSIMDPP_DISPATCH_ARCH2=SIMDPP_ARCH_X86_SSE3 \
        -DSIMDPP_DISPATCH_ARCH3=SIMDPP_ARCH_X86_SSE4_1 \
        -DSIMDPP_DISPATCH_ARCH4=SIMDPP_ARCH_X86_AVX \
        -DSIMDPP_DISPATCH_ARCH5=SIMDPP_ARCH_X86_AVX2 \
        -o test_sse2.o
g++ test.cc -c -std=c++11 -I../../ -DSIMDPP_ARCH_X86_SSE3 -msse3 -o test_sse3.o
g++ test.cc -c -std=c++11 -I../../ -DSIMDPP_ARCH_X86_SSE4_1 -msse4.1 -o test_sse4_1.o
g++ test.cc -c -std=c++11 -I../../ -DSIMDPP_ARCH_X86_AVX -mavx -o test_avx.o
g++ test.cc -c -std=c++11 -I../../ -DSIMDPP_ARCH_X86_AVX2 -mavx2 -o test_avx2.o
g++ main.o test_sse2.o test_sse3.o test_sse4_1.o test_avx.o test_avx2.o -o test
test_sse2.o: In function `print_arch()':
test.cc:(.text+0x69): undefined reference to `arch_null::register_fn_print_arch(void (*)())'
collect2: error: ld returned 1 exit status
```

**Comment by p12tic** [comment]
Hmm, I was testing a bit more elaborate example locally. I can reproduce with the version in the repo --  sorry for the inconvenience. I'll push a fix once the testsuite has finished running.

**Comment by p12tic** [comment]
Fixed in 01b3edd80605f.
